test/support/contexts.rb

```diff
@@ -20,23 +20,32 @@ def with_hidden_elements
 def with(params = {})
   factory = params[:factory]
   factories = params[:factories] || []
-  js_driver = if params[:js] == :deprecated
-                :poltergeist_custom
-              else
-                (ENV["INTEGRATION_TEST_DRIVER"] || :headless_chrome).to_sym
-              end
+  js_driver = js_driver(params[:js])
+  admin = params[:admin]
 
   Capybara.current_driver = js_driver if params[:js]
+  sign_in_admin(admin) if admin
 
   if (site = params[:site])
     stub_current_site(site) { yield }
   else
     yield(params)
   end
 
+  sign_out_admin if admin
```

Comment:
I think we have to move

```ruby
sign_out_admin if admin
Capybara.reset_session! if params[:js]
```

to the `ensure` block to avoid keeping the admin logged in for the following tests if the current tests fails

Reply:
Done!